index.js

```diff
@@ -61,27 +61,47 @@ async function run() {
     const getTeamMembers = (team) => {
       if (team.formation) {
         return team.formation.reduce((allMembers, formationTeam) => {
-          const team = raw.teams.find(team => team.name === formationTeam);
+          const team = raw.teams.find((team) => team.name === formationTeam);
           return allMembers.concat(getTeamMembers(team));
         }, []);
       }
       return [...(team.members || []), ...(team.maintainers || [])];
-    }
+    };
 
-    // Check that each GitHub team has a maintainer.
     for (const team of raw.teams) {
+      // Check that each GitHub team has a maintainer.
       if (!team.formation && !hasMaintainers(team)) {
         core.setFailed(
           `GitHub team ${team.name} does not have valid maintainer(s)`,
         );
         return;
       }
 
+      // Check that maintainers are not duplicated in the members field
+      if (hasMaintainers(team)) {
+        const maintainerAndMember = team.maintainers.filter((maintainer) => {
```

Comment:
perhaps a bit simpler here to use `Array.some()`?

```
if (team.maintainers.some(m => team.members.includes(m))) {
  core.setFailed(
    `GitHub team ${
      team.name
    } has one or more maintainers who are also duplicated as members: ${maintainerAndMember.join(
      ', ',
    )}`,
  );
  return;
}
```

Reply:
👀 I had that originally but the output is clearer this way because we collect the member names!